src/Symfony/Component/Mailer/Mailer.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer;
+
+use Symfony\Component\Mailer\Messenger\EnvelopedMessage;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Mime\RawMessage;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 4.3
+ */
+class Mailer implements MailerInterface
+{
+    private $transport;
+    private $bus;
+
+    public function __construct(TransportInterface $transport, MessageBusInterface $bus = null)
```

Comment:
looks like 2 different `Mailer` implementations to me

Reply:
one returns something and is thus synchronous: the transport.
the mailer sends but and doesn't return anything, thus is compatible with async buses: the mailer

Reply:
ok, i've read the MailerInterface class description now :) :+1: